### PR TITLE
[AC-2576] Replace Billing commands and queries with services

### PR DESCRIPTION
**Pull request by amorask-bitwarden**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

I wrote this PR due to ongoing conversations in the #code channel regarding how Bitwarden handles "CQRS" and the issue of having commands invoke other commands. Since Billing as a business function is more of an intermediary layer that processes side effects of other user actions, we've decided to refactor the business logic we've been writing to be service-oriented rather than formatted as commands or queries.

(Sorry for the insane commit chain. I've outlined the changes by team below)

This resulted in the movement of the following logic:
- `AssignSeatsToClientOrganizationCommand` -> `Commercial.ProviderBillingService`
- `CancelSubscriptionCommand` -> `SubscriberService`
- `CreateCustomerCommand` -> `Commercial.ProviderBillingService`
- `RemovePaymentMethodCommand` -> `SubscriberService`
- `ScaleSeatsCommand` -> `Commercial.ProviderBillingService`
- `StartSubscriptionCommand` -> `Commercial.ProviderBillingService`
- `ProviderBillingQueries` -> `Commercial.ProviderBillingService`
- `OrganizationBillingQueries` -> `OrganizationBillingService`
- `SubscriberQueries` -> `SubscriberService`

@bitwarden/team-admin-console-dev:
As far as changes to your domain go, the big one is in `RemoveOrganizationFromProviderCommand`. Since `RemovePaymentMethod` is no longer a command, we can now invoke it from its service within this command. That update is in this PR and I've added tests for it.

The updates to your API Controllers are just wiring up the new services with the same logic instead of the commands and queries. The only actual change to billing business logic represented in AC code is in the API `ProvidersController`. You'll see `StartSubscriptionCommand` has been split into 2 methods: `CreateCustomer` and `StartSubscription`. They do the same thing as before, but I wanted method control to be more granular.

@bitwarden/team-auth-dev:
I think you only have one file changed in here and it's just replacing the previous command invocation for cancelling a subscription with the new service method; should be pretty straight forward.

@bitwarden/team-billing-dev: 
If needed, we can sync as a team on the changes. Jet let me know.

### CODE REVIEW

File: bitwarden_license/test/Commercial.Core.Test/AdminConsole/ProviderFeatures/RemoveOrganizationFromProviderCommandTests.cs [hunk 1]
```diff
@@ -85,161 +86,154 @@ public class RemoveOrganizationFromProviderCommandTests
...
+                [],
                 includeProvider: false)
             .Returns(true);
 
-        var organizationOwnerEmails = new List<string> { "a@gmail.com", "b@gmail.com" };
+        var organizationRepository = sutProvider.GetDependency<IOrganizationRepository>();
 
-        organizationRepository.GetOwnerEmailAddressesById(organization.Id).Returns(organizationOwnerEmails);
+        organizationRepository.GetOwnerEmailAddressesById(organization.Id).Returns([
+            "a@b.com",
```
Comment (r-tome) [inline bitwarden_license/test/Commercial.Core.Test/AdminConsole/ProviderFeatures/RemoveOrganizationFromProviderCommandTests.cs:109]: Nit: I prefer to use the domain `example.com` that is reserved for testing purposes

**Review by r-tome — DISMISSED** [review]
AC changes look good!

**Review by jlf0dev — APPROVED** [review]

**Review by cturnbull-bitwarden — DISMISSED** [review]

**Comment by amorask-bitwarden** [comment]
@jlf0dev @r-tome @cturnbull-bitwarden Hey everyone, sorry had to fix something up in the `RemoveOrganizationFromProviderCommand`. That change is represented here: https://github.com/bitwarden/server/pull/4070/commits/734e77ce36cc622f4a9b3a2e4d8609305395848c

If I could get another look, would greatly appreciate it.

**Review by jlf0dev — APPROVED** [review]
Auth looks good 👍

**Review by r-tome — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by r-tome — DISMISSED** [review]

**Review by cturnbull-bitwarden — DISMISSED** [review]

**Review by r-tome — APPROVED** [review]

**Review by cturnbull-bitwarden — DISMISSED** [review]

**Review by r-tome — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**